### ISSUE TITLE
TLS used on debug build even if disabled?

### ISSUE THREAD

**Issue by davidneilson1337** [gabime/spdlog#613]
Hi,

I'm using spdlog in my application, which cannot support TLS.

I'm building spdlog with the SPDLOG_DISABLE_TID_CACHING flag.
When using spdlog on release builds, all of it's components work flawlessly.
However, when using spdlog on debug builds, calling spdlog functions or creating logger objects throws a SEH exception, probably indicating that TLS is used by the library.

I would appreciate some help on the topic, perhaps its not about thread local storage but some other thing i'm missing?

Many thanks,
David

**Comment by gabime** [comment]
Which compiler and platform are you using? Why do you think the SEH thrown is related to the TLS ?

**Comment by davidneilson1337** [comment]
Thanks for the quick reply :)
I am using MSVC14.0 (VS2015) on windows 10 x64. I also compiled using MSVC14.1 (VS2017) with the same results.
My program throws the exact same exception either when I use TLS manually in my code or when I link against libraries which state they use TLS.
I'm compiling my program with nothing else than creating a standard (either st / mt) stdout logger in the main function. 
Also, just to make sure I compiled an empty main, and it didn't throw.
